### PR TITLE
Reset LiveVersionMap on sync commit

### PR DESCRIPTION
**Pull request by s1monw**
Today we carry on the size of the live version map to ensure that
we minimze rehashing. Yet, once we are idle or we can issue a sync-commit
we can resize it to defaults to free up memory.

Relates to #27516

### CODE REVIEW

**Comment by s1monw** [comment]
@bleskes ping

**Review by bleskes — APPROVED** [review]
LGTM. Yay for the tests

File: core/src/test/java/org/elasticsearch/index/engine/LiveVersionMapTests.java [hunk 1]
```diff
@@ -57,4 +70,144 @@ public void testRamBytesUsed() throws Exception {
...
+            });
+            if (randomBoolean()) {
+                Thread.yield();
+            }
+        } while (done.getCount() != 0);
+
+        for (int j = 0; j < threads.length; j++) {
+            threads[j].join();
+        }
+        map.getAllCurrent().forEach((k, v) -> {
```
Comment (bleskes) [inline core/src/test/java/org/elasticsearch/index/engine/LiveVersionMapTests.java]: can we add a check that all deletes are tracked in the tombstone?

Comment (s1monw) [reply core/src/test/java/org/elasticsearch/index/engine/LiveVersionMapTests.java]: DONE!

Comment (bleskes) [reply core/src/test/java/org/elasticsearch/index/engine/LiveVersionMapTests.java]: THANKS! ;)

**Comment by floragunncom** [comment]
this caused #27852

**Pull request merged**